proxy/enfuse/server.go

```diff
@@ -0,0 +1,102 @@
+package enfuse
+
+import (
+	"context"
+	"fmt"
+	enfuse "github.com/enfabrica/enkit/proxy/enfuse/rpc"
+	"google.golang.org/grpc"
+	"io"
+	"io/ioutil"
+	"net"
+	"os"
+	"path/filepath"
+)
+
+func ServeDirectory(mods ...ServerConfigMod) error {
+	s := NewServer(NewServerConfig(mods...))
+	return s.Serve()
+}
+
+var _ enfuse.FuseControllerServer = &FuseServer{}
+
+type FuseServer struct {
+	cfg *ServerConfig
+}
+
+func (s *FuseServer) SingleFileInfo(ctx context.Context, request *enfuse.SingleFileInfoRequest) (*enfuse.SingleFileInfoResponse, error) {
+	des, err := os.Open(filepath.Join(s.cfg.Dir, request.Path))
+	if err != nil {
+		return nil, err
+	}
+	defer des.Close()
+	st, err := des.Stat()
+	if err != nil {
+		return nil, err
+	}
+	return &enfuse.SingleFileInfoResponse{
+		Info: &enfuse.FileInfo{
+			Name:  filepath.Base(request.Path),
+			Size:  st.Size(),
+			IsDir: false,
+		},
+	}, nil
+}
+
+func (s *FuseServer) Files(ctx context.Context, rf *enfuse.RequestFile) (*enfuse.ResponseFile, error) {
+	f, err := os.Open(filepath.Join(s.cfg.Dir, rf.Path))
+	if err != nil {
+		return nil, err
+	}
+	defer f.Close()
+	data := make([]byte, rf.Size) // default to sending 1mb at max, client side is 4mb max possible but this is to be safe
```

Comment:
Where is this max enforced?

Reply:
It's a limitation set by grpc itself (it's totally arbitrary)